pkg/subctl/cmd/deploybroker.go

```diff
@@ -49,7 +49,15 @@ func init() {
 	deployBroker.PersistentFlags().StringVar(&ipsecSubmFile, "ipsec-psk-from", "",
 		"Import IPsec PSK from existing submariner broker file, like broker-info.subm")
 
-	deployBroker.PersistentFlags().BoolVar(&serviceDiscovery, "service-discovery", false,
+	deployBroker.PersistentFlags().Bool("service-discovery", true,
+		"Enable Multi Cluster Service Discovery")
+
+	err := deployBroker.Flags().MarkDeprecated("service-discovery",
```

Comment:
Just a thought: Instead of deprecating the flag "service-discovery", why not just modify the default from false to true?

Reply:
It's a cmdline flag :-) then you can't disable it, how do you disable it with --service-discovery anymore? :)

Reply:
You can do `--service-discovery=false`

Reply:
Not sure why my comment didn't go through.

We can only deprecate it because

subctl --service-discovery being called doesn't make sense anymore (is a bool flag, does not accept true/false afterwards,...)

Reply:
As Vishal says, `--service-discovery` is still useful to disable service discovery; boolean flags [support a value after an equals symbol](https://github.com/spf13/pflag/#command-line-flag-syntax).

Reply:
> You can do `--service-discovery=false`

can you? ? let me try